airbyte-integrations/connectors/destination-langchain/destination_langchain/embedder.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from destination_langchain.config import FakeEmbeddingConfigModel, OpenAIEmbeddingConfigModel
+from langchain.embeddings.base import Embeddings
+from langchain.embeddings.fake import FakeEmbeddings
+from langchain.embeddings.openai import OpenAIEmbeddings
+
+
+class Embedder(ABC):
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def check(self) -> Optional[str]:
+        pass
+
+    @property
+    @abstractmethod
+    def langchain_embeddings(self) -> Embeddings:
+        pass
+
+    @property
+    @abstractmethod
+    def embedding_dimensions(self) -> int:
+        pass
+
+
+OPEN_AI_VECTOR_SIZE = 1536
+
+
+class OpenAIEmbedder(Embedder):
+    def __init__(self, config: OpenAIEmbeddingConfigModel):
+        super().__init__()
+        self.embeddings = OpenAIEmbeddings(openai_api_key=config.openai_key, chunk_size=8191)
+
+    def check(self) -> Optional[str]:
+        try:
+            self.embeddings.embed_query("test")
+        except Exception as e:
+            return str(e)
+        return None
+
+    @property
+    def langchain_embeddings(self) -> Embeddings:
+        return self.embeddings
+
+    @property
+    def embedding_dimensions(self) -> int:
+        # vector size produced by text-embedding-ada-002 model
```

Comment:
Should the vector size and/or model be configurable? Is `text-embedding-ada-002` the default? I'm not seeing it mentioned anywhere else.

Reply:
It is mentioned in the documentation and as a description as part of the spec: 
* https://github.com/airbytehq/airbyte/blob/0a571f10d870cf4b5deb394a476c194b8fae01da/airbyte-integrations/connectors/destination-langchain/destination_langchain/config.py#L44
* https://github.com/airbytehq/airbyte/blob/0a571f10d870cf4b5deb394a476c194b8fae01da/docs/integrations/destinations/langchain.md#L10

I think it makes sense to start like this (OpenAI recently deprecated all other embedding models as this one is preferable in all situations), we probably want to add other embeddings at a later stage though that might have different numbers of dimensions (the number of dimensions is an intrinsic part of the embedding model).